src/pages/index.js

```diff
@@ -53,45 +63,220 @@ const addFormValidator = new FormValidator(
+    })
+    .finally(() => {
+      profileEditSaveButton.textContent = "Save";
+      profileEditSaveButton.classList.remove("loading");
+      profileEditSaveButton.disabled = false;
+    });
 }
 
+const handleLikeClick = (card) => {
+  api
+    .toggleLike(card.getId(), card.isLiked())
+    .then(() => {
+      card.updateLikesView();
+    })
+    .catch((err) =>
+      console.error(`An error occurred when changing like status: ${err}`)
+    );
+};
+
 function handleAddCardFormSubmit(data) {
-  const newCardElement = createCard({ name: data.title, link: data.url });
-  section.addItem(newCardElement);
-  addCardFormElement.reset();
-  addFormValidator.disableButton();
-  addCardPopup.close();
+  addCardSaveButton.textContent = "Saving...";
+  addCardSaveButton.classList.add("loading");
+  addCardSaveButton.disabled = true;
+  const sanitizedUrl = data.url.replace(/["'\s]+$/, "").trim();
+  api
+    .addCard(data.title, sanitizedUrl)
+    .then((newCardInfo) => {
+      const newCardElement = createCard({
+        name: newCardInfo.name,
+        link: newCardInfo.link,
+        _id: newCardInfo._id,
+      });
+      section.addItem(newCardElement);
+      addCardFormElement.reset();
+      addFormValidator.resetValidation();
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error adding new card:", err);
+    })
+    .finally(() => {
+      addCardSaveButton.textContent = "Save";
+      addCardSaveButton.classList.remove("loading");
+      addCardSaveButton.disabled = false;
+    });
+}
+
+function handleDeleteConfirm(data) {
+  const cardId = data.cardId;
+  if (!cardId || typeof cardId !== "string") {
+    console.error(
+      "Invalid cardId in handleDeleteConfirm:",
+      cardId,
+      "Data:",
+      data
+    );
+    return;
+  }
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      section.removeItem(cardId);
+      deleteCardPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error deleting card:", err);
+    });
+}
+
+function handleAvatarFormSubmit(data) {
+  avatarSubmitButton.textContent = "Saving...";
+  avatarSubmitButton.classList.add("loading");
+  avatarSubmitButton.disabled = true;
+  api
+    .updateUserAvatar(data.avatar)
+    .then((updatedInfo) => {
+      userInfo.setUserInfo({
+        name: updatedInfo.name,
+        about: updatedInfo.about,
+        avatar: updatedInfo.avatar,
+      });
+      avatarEditPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar", err);
+    })
+    .finally(() => {
+      avatarSubmitButton.textContent = "Save";
+      avatarSubmitButton.classList.remove("loading");
+      avatarSubmitButton.disabled = false;
+    });
 }
 
+avatarEditButton.addEventListener("click", () => {
+  const currentUserInfo = userInfo.getUserInfo();
+  avatarInput.value = currentUserInfo.avatar || "";
+  avatarEditPopup.open();
+  avatarFormValidator.resetValidation();
+  avatarFormValidator.enableValidation();
```

Comment:
`  avatarFormValidator.enableValidation();
`

Please, delete it

`enableValidation` should be called only 1 time to enable validation. You should not do that when you open/close a popup because you add more and more the same listeners to the same forms. This may cause a `memory leak`